code/playground/src/main/scala/com/avast/datadog4s/playground/CreateMetric.scala

```diff
@@ -1,31 +1,52 @@
 package com.avast.datadog4s.playground
 
-import cats.effect.{ ExitCode, IO, IOApp }
+import cats.Monad
+import cats.effect.kernel.Ref
+import cats.effect.{ ExitCode, IO, IOApp, Resource }
+import com.avast.datadog4s.api.MetricFactory
 import com.avast.datadog4s.api.metric.{ Distribution, Histogram }
+import com.avast.datadog4s.extension.jvm.JvmMonitoring
 import com.avast.datadog4s.{ StatsDMetricFactory, StatsDMetricFactoryConfig }
+import scala.collection.compat.immutable._
 
 import java.net.InetSocketAddress
 
 object CreateMetric extends IOApp {
-  override def run(args: List[String]): IO[ExitCode] =
-    StatsDMetricFactory
-      .make[IO](StatsDMetricFactoryConfig(Some("playground"), new InetSocketAddress("127.0.0.1", 8125)))
-      .use { factory =>
-        for {
-          _   <- IO.unit
-          dist = factory.distribution.long("distribution1")
-          hist = factory.histogram.long("histogram1")
-          _   <- loop(hist, dist)
-        } yield ExitCode.Success
-      }
-  def loop(hist: Histogram[IO, Long], dist: Distribution[IO, Long]): IO[Unit] = {
-    val l = scala.util.Random.nextInt(100)
-    import scala.concurrent.duration._
-    val p = for {
-      _ <- hist.record(l.toLong)
-      _ <- dist.record(l.toLong)
-      _ <- IO.sleep(100.milli)
+  override def run(args: List[String]): IO[ExitCode] = {
+    val task = for {
+      factory <- StatsDMetricFactory
+                   .make[IO](StatsDMetricFactoryConfig(Some("playground"), new InetSocketAddress("127.0.0.1", 8125)))
+      _       <- JvmMonitoring.default(factory)
+      result  <- Resource.eval(doRun(factory))
+    } yield result
+    task.use(_ => IO.pure(())).race(IO.readLine.attempt.flatMap(_ => IO.pure(()))).map(_ => ExitCode.Success)
```

Comment:
What are you trying to do here?

Reply:
basically wait until `enter` is pressed to kill the program

Reply:
So, can you please use `IO.unit` instead of `IO.pure(())` and `readline.attempt.void` instead of `.flatMap(_ => IO.pure(())` ?

Reply:
👍